swiftclient/api.go

```diff
@@ -109,8 +127,8 @@ func ObjectDelete(accountName string, containerName string, objectName string, o
 }
 
 // ObjectFetchChunkedPutContext provisions a context to use for an HTTP PUT using "chunked" Transfer-Encoding on the named Swift Object.
-func ObjectFetchChunkedPutContext(accountName string, containerName string, objectName string) (chunkedPutContext ChunkedPutContext, err error) {
-	return objectFetchChunkedPutContextWithRetry(accountName, containerName, objectName)
+func ObjectFetchChunkedPutContext(accountName string, containerName string, objectName string, useReserveForVolumeName ...string) (chunkedPutContext ChunkedPutContext, err error) {
+	return objectFetchChunkedPutContextWithRetry(accountName, containerName, objectName, useReserveForVolumeName...)
 }
```

Comment:
There's no good reason to make this a variadic function.  If we need to add a "volume name" parameter then just add an additional paramater and be done with it.

ObjectFetchChunkedPutContext() is only called in 19 places.  Updating the 18 of them not currently affected to add ", nil" to the end of the function call would take less time then we're probably going to spend arguing about this, particularly since the compiler will flag any places we missed.

Assuming we go forward with this change, I'd be happy to go through and add the nil argument to the other 18 places and strip out code for the variadic arguments.

Reply:
Happy to make this change to remove the need for the variadic... it'll also make the comparison for the option's existence more clear. @bschatz-swift earlier made a comment about "why are you taking the length of the arg?" before... not very intuitive to be sure. But, then, variadics in positional argument lists have always been, uh, a hack imho :-).